src/zx_plot.jl

```diff
@@ -0,0 +1,208 @@
+                            frontier_v[qubit_loc(lo, u)] += 1
+                        end
+                        for q in min(qubit_loc(lo, v), qubit_loc(lo, v1)):max(qubit_loc(lo, v), qubit_loc(lo, v1))
+                            frontier_locs[q] = x + 1
+                        end
+                    end
+                end
+            end
+        end
+    end
+    for v in vs
+        if !haskey(locs, v)
+            v1, v2 = neighbors(zxd, v)
+            x1, y1 = locs[v1]
+            x2, y2 = locs[v2]
+            locs[v] = ((x1+x2)/2, (y1+y2)/2)
+        end
+    end
+    locs_x = [locs[v][1] for v in vs]
+    locs_y = [locs[v][2] for v in vs]
+    return locs_x, locs_y
+end
+
+function layout2locs(zxd::ZXGraph{T,P}) where {T,P}
+    lo = zxd.layout
+    vs = spiders(zxd)
+    locs = Dict()
+    nqubit = lo.nbits
+    frontier_v = ones(T, nqubit)
+    frontier_locs = ones(nqubit)
+    phase_gadget_loc = 1.0
+
+    for v in vs
+        if qubit_loc(lo, v) != nothing
+            y = qubit_loc(lo, v)
+            x = findfirst(isequal(v), lo.spider_seq[y])
+            locs[v] = (Float64(x), Float64(y))
+        else
+            locs[v] = nothing
+        end
+    end
+    for v in vs
+        if locs[v] == nothing
+            nb = neighbors(zxd, v)
+            if length(nb) == 1
+                u = nb[1]
+                locs[v] = (phase_gadget_loc, Float64(nqubit + 2))
+                locs[u] = (phase_gadget_loc, Float64(nqubit + 1))
+                phase_gadget_loc += 1
+            end
+
+            # v1, v2 = neighbors(zxd, v)
+            # x1, y1 = locs[v1]
+            # x2, y2 = locs[v2]
+            # locs[v] = ((x1+x2)/2, (y1+y2)/2)
+        end
+    end
+    println(locs)
+    locs_x = [locs[v][1] for v in vs]
+    locs_y = [locs[v][2] for v in vs]
+    return locs_x, locs_y
+end
+
+function ZXplot(zxd::ZXDiagram; linetype = "straight")
+    g, edgelabel = ZX2Graph(zxd)
+    nodelabel = ZX2nodelabel(zxd)
+    nodefillc = ZX2nodefillc(zxd)
+    edgelabelc = colorant"black"
+    if zxd.layout.nbits > 0
+        locs_x, locs_y = layout2locs(zxd)
+        size_x = maximum(locs_x) - minimum(locs_x)
+        size_y = maximum(locs_y) - minimum(locs_y)
+        set_default_graphic_size(3size_x*cm, 3size_y*cm)
+        composition = gplot(g,
+            locs_x, locs_y,
+            nodelabel = nodelabel, edgelabel = edgelabel, edgelabelc = edgelabelc, nodefillc = nodefillc,
+            linetype = linetype,
+            NODESIZE = 1/(2size_x),
+            # EDGELINEWIDTH = 8.0 / sqrt(nv(g))
+        )
+        # draw(SVG("test.svg", size_x*cm, size_y*cm), composition)
+    else
+        gplot(g,
+            nodelabel = nodelabel, edgelabel = edgelabel, edgelabelc = edgelabelc, nodefillc = nodefillc,
+            linetype = linetype,
+            # NODESIZE = 0.35 / sqrt(nv(g)), EDGELINEWIDTH = 8.0 / sqrt(nv(g))
+        )
+    end
+end
+function ZXplot(zxd::ZXGraph; linetype = "straight")
```

Comment:
we can just use `plot` here, since there is a type already, so we could simplify the interface a lot in the future via dispatch.

Reply:
don't worry for conflicts, we are not going to use other general plotting packages anyway.